src/core/models.py

```diff
@@ -1919,3 +1968,641 @@ def log_hijack_ended(sender, hijacker, hijacked, request, **kwargs):
 
 hijack_started.connect(log_hijack_started)
 hijack_ended.connect(log_hijack_ended)
+
+
+class OrganizationName(models.Model):
+    value = models.CharField(
+        max_length=1000,
+        verbose_name="Organization name",
+    )
+    ror_display_for = models.OneToOneField(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='ror_display',
+        blank=True,
+        null=True,
+        help_text="This name is a preferred ROR-provided display name.",
+    )
+    label_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='labels',
+        blank=True,
+        null=True,
+        help_text="This name is a preferred ROR-provided alternative, "
+            "often in a different language from the ROR display name.",
+    )
+    custom_label_for = models.OneToOneField(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='custom_label',
+        blank=True,
+        null=True,
+        help_text="This name is a custom label entered by the end user. "
+            "Only exists in Janeway, independent of ROR.",
+    )
+    alias_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='aliases',
+        blank=True,
+        null=True,
+        help_text="This name is a less preferred ROR-provided alternative.",
+    )
+    acronym_for = models.ForeignKey(
+        'Organization',
+        on_delete=models.CASCADE,
+        related_name='acronyms',
+        blank=True,
+        null=True,
+        help_text="This name is a ROR-provided acronym.",
+    )
+    language = models.CharField(
+        max_length=10,
+        blank=True,
+        choices=submission_models.LANGUAGE_CHOICES,
+    )
```

Comment:
Is there a rationale behind avoiding the use of `django-modeltranslations` here? Does it have to do with the need for a distinction between translations and `ror_display_for`?

Reply:
Thanks for raising this. Yes, ROR data sometimes has multiple names per language. I've written it up here: https://github.com/openlibhums/janeway/issues/4657

Coincidentally, the importer was bugged because ROR gives 2-character language codes but our `LANGUAGE_CHOICES` uses 3-character codes. So I've fixed that by using a small ISO-639 utility. I've also documented which version of ISO-639 we hardcode, for future reference.